pkg/checkmarx/checkmarx.go

```diff
@@ -0,0 +1,469 @@
+	"net/http"
+	"net/url"
+	"strconv"
+
+	"github.com/SAP/jenkins-library/pkg/log"
+	piperHttp "github.com/SAP/jenkins-library/pkg/http"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+// AuthToken - Structure to store OAuth2 token
+type AuthToken struct {
+	TokenType   string `json:"token_type"`
+	AccessToken string `json:"access_token"`
+	ExpiresIn   int    `json:"expires_in"`
+}
+
+// Preset - Project's Preset
+type Preset struct {
+	ID        int    `json:"id"`
+	Name      string `json:"name"`
+	OwnerName string `json:"ownerName"`
+	Link      Link   `json:"link"`
+}
+
+// Scan - Scan Structure
+type Scan struct {
+	ID   int  `json:"id"`
+	Link Link `json:"link"`
+}
+
+// ResultsStatistics - ResultsStatistics Structure
+type ResultsStatistics struct {
+	High   int `json:"highSeverity"`
+	Medium int `json:"mediumSeverity"`
+	Low    int `json:"lowSeverity"`
+	Info   int `json:"infoSeverity"`
+}
+
+// ScanStatus - ScanStatus Structure
+type ScanStatus struct {
+	ID       int    `json:"id"`
+	Link     Link   `json:"link"`
+	Status   Status `json:"status"`
+	ScanType string `json:"scanType"`
+	Comment  string `json:"comment"`
+}
+
+// Status - Status Structure
+type Status struct {
+	ID   int    `json:"id"`
+	Name string `json:"name"`
+}
+
+// Project - Project Structure
+type Project struct {
+	ID                 int                `json:"id"`
+	TeamID             string             `json:"teamId"`
+	Name               string             `json:"name"`
+	IsPublic           bool               `json:"isPublic"`
+	SourceSettingsLink SourceSettingsLink `json:"sourceSettingsLink"`
+	Link               Link               `json:"link"`
+}
+
+// Team - Team Structure
+type Team struct {
+	ID       string `json:"id"`
+	FullName string `json:"fullName"`
+}
+
+// Link - Link Structure
+type Link struct {
+	Rel string `json:"rel"`
+	URI string `json:"uri"`
+}
+
+// SourceSettingsLink - SourceSettingsLink Structure
+type SourceSettingsLink struct {
+	Type string `json:"type"`
+	Rel  string `json:"rel"`
+	URI  string `json:"uri"`
+}
+
+// Checkmarx is the client communicating with the Checkmarx backend
+type Checkmarx struct {
+	serverURL string
+	client    piperHttp.Client
+	logger    *logrus.Entry
+}
+
+// NewCheckmarx returns a new Checkmarx client for communicating with the backend
+func NewCheckmarx(serverURL, username, password string ) (*Checkmarx, error) {
+	cmx := Checkmarx {
+		serverURL: serverURL,
+		logger: log.Entry().WithField("package", "SAP/jenkins-library/pkg/checkmarx"),
+		client: piperHttp.Client {},
+	}
+
+	token, err := getOAuth2Token(serverURL, username, password)
+	if err != nil {
+		return &cmx, errors.Wrap(err, "error fetching oAuth token")
+	}
+
+	options := piperHttp.ClientOptions {
+		Token: token,
+	}
+	
+	cmx.client.SetOptions(options)
+	
+	return &cmx, nil
+}
+
+func getOAuth2Token(serverURL, username, password string) (string, error) {
+	resp, err := http.PostForm(serverURL+"auth/identity/connect/token",
```

Comment:
I guess we should rather use the piperhttp Client here.
This makes it easier to test ...